brave/brave-core
# Omnibox Command Palette

Resolves https://github.com/brave/brave-browser/issues/28178

Adds a prototype command palette in the Omnibox. Behind the Chromium brave://flags/#quick-commands flag. Note: This flag is set to expire in M111 so you might need to also enable the brave://flags/#temporary-unexpire-flags-m110 flag if you're running a recent version of Brave.

The crbug issue for this feature is here: https://bugs.chromium.org/p/chromium/issues/detail?id=1014639

[commander-demo.webm](https://user-images.githubusercontent.com/7678024/216176461-1eb5d43e-0d9a-4045-ad51-2de4e9e93a1b.webm)

## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

1. Enable the `brave://flags#quick-commands` flag (if not present enable the `Temporarily Enable M110 flags`)
2. Press <kbd>Ctrl</kbd>+<kbd>Space</kbd>
3. The Omnibox should focus, and a list of available commands should be visible
4. Type `Create tab` and select the `Create new tab` command. A new tab should open
5. Press <kbd>Ctrl</kbd>+<kbd>Space</kbd>
6. Type `Move window` and select `Move tab to new window`. The tab should move to a new window
7. Navigate to `https://readr.nz`
8. Press <kbd>Ctrl</kbd>+<kbd>Space</kbd>
9. Type `Move window` and select `Move tabs to window...`
10. Select the `New tab` (or whatever your original window was called).
11. The readr.nz tab should move back to the original window.
